libnmstate/state.py

```diff
@@ -30,6 +30,43 @@
 from libnmstate.prettystate import format_desired_current_state_diff
 from libnmstate.schema import Interface
 from libnmstate.schema import InterfaceType
+from libnmstate.schema import Route
+
+
+class _Route(object):
+    def __init__(self, route):
+        self.state = copy.deepcopy(route)
+        if route.get(Route.STATE) != Route.ABSENT:
+            if Route.TABLE_ID not in route:
+                self.state[Route.TABLE_ID] = Route.USE_DEFAULT_ROUTE_TABLE
+            if Route.METRIC not in route:
+                self.state[Route.METRIC] = Route.USE_DEFAULT_METRIC
+            if Route.NEXT_HOP_ADDRESS not in route:
+                self.state[Route.NEXT_HOP_ADDRESS] = ''
+
+    def __hash__(self):
+        return hash((self.state.get(Route.STATE),
+                     self.state.get(Route.TABLE_ID),
+                     self.state.get(Route.DESTINATION),
+                     self.state.get(Route.NEXT_HOP_INTERFACE),
+                     self.state.get(Route.NEXT_HOP_ADDRESS),
+                     self.state.get(Route.METRIC)))
+
+    def __eq__(self, other):
+        return self.state == other.state
```

Comment:
Usually an identity check is done in such cases for optimization.

Reply:
Can you elaborate on this?